File: backend/app/repositories/navieras.py
"""Repositorio para navieras"""
from sqlalchemy.orm import Session
from typing import List, Optional
from ..models import Naviera
from ..schemas import NavieraCreate, NavieraUpdate


class NavieraRepository:
    """Repositorio para operaciones CRUD de navieras"""
    
    def get_all(self, db: Session) -> List[Naviera]:
        """Obtener todas las navieras"""
        return db.query(Naviera).order_by(Naviera.nombre).all()
    
    def get_by_id(self, db: Session, id_navieras: int) -> Optional[Naviera]:
        """Obtener naviera por ID"""
        return db.query(Naviera).filter(Naviera.id_navieras == id_navieras).first()
    
    def get_by_codigo(self, db: Session, codigo: str) -> Optional[Naviera]:
        """Obtener naviera por código"""
        return db.query(Naviera).filter(Naviera.codigo == codigo).first()
    
    def create(self, db: Session, naviera_data: NavieraCreate) -> Naviera:
        """Crear nueva naviera"""
        naviera = Naviera(**naviera_data.model_dump())
        db.add(naviera)
        db.commit()
        db.refresh(naviera)
        return naviera
    
    def update(self, db: Session, naviera: Naviera, naviera_data: NavieraUpdate) -> Naviera:
        """Actualizar naviera existente"""
        update_data = naviera_data.model_dump(exclude_unset=True)
        for key, value in update_data.items():
            setattr(naviera, key, value)
        db.commit()
        db.refresh(naviera)
        return naviera
    
    def delete(self, db: Session, naviera: Naviera) -> None:
        """Eliminar naviera"""
        db.delete(naviera)
        db.commit()


naviera_repository = NavieraRepository()

File: frontend/src/api/navieras.ts
/**
 * Servicio API para navieras
 */
import axios from './axios';
import type { Naviera, NavieraCreate, NavieraUpdate, ApiMessage } from '../types';

export const navierasApi = {
  /**
   * Obtener todas las navieras
   */
  listar: async (): Promise<Naviera[]> => {
    const response = await axios.get<Naviera[]>('/navieras');
    return response.data;
  },

  /**
   * Crear nueva naviera
   */
  crear: async (data: NavieraCreate): Promise<Naviera> => {
    const response = await axios.post<Naviera>('/navieras', data);
    return response.data;
  },

  /**
   * Actualizar naviera
   */
  actualizar: async (id: number, data: NavieraUpdate): Promise<Naviera> => {
    const response = await axios.put<Naviera>(`/navieras/${id}`, data);
    return response.data;
  },

  /**
   * Eliminar naviera
   */
  eliminar: async (id: number): Promise<ApiMessage> => {
    const response = await axios.delete<ApiMessage>(`/navieras/${id}`);
    return response.data;
  },
};
